hw2/03-count.js

```diff
@@ -1,3 +1,24 @@
-// Add your code here
+//Return the source word, with html to highlight it if it matches the target word
+const replaceWithHighlightHTML = (source, target) => {
+  return source === target
+    ? `<mark class='bg-warning'>${target}</mark>`
+    : source;
+};
 
-input.addEventListener('keydown', handleKeyDown);
+//Highlight words in the paragraph based on user input
+const handleKeyUp = () => {
+  const input_value = document
+    .getElementById("userInput")
+    .value.trim()
+    .toLowerCase();
+  const search_element = document.getElementById("toSearch");
+  const search_arr = search_element.textContent
+    .toLowerCase()
+    .split(RegExp("\\b"));
+
+  search_element.innerHTML = search_arr
```

Comment:
**Best Practices**
Directly setting **innerHTML** can pose a security risk if not handled correctly, especially with user input. It is better to use **textContent**.

Reply:
In this context this can't be used maliciously, so for the sake of time I'm leaving it. I will bear this in mind in the future though, thanks.